### ISSUE TITLE
resources { 'sensu_check': purge => true; } gives unexpected result

### ISSUE THREAD

**Issue by maxadamo** [sensu/sensu-puppet#1157]
## Description of problem

- What did you do?

added to my manifest this single line: 
```
resources { 'sensu_check': purge => true; }
```

If I comment this line out, everything works perfectly. 

## Command used and debugging output

please read subsequent messages in this issue. 

## Platform and version information

- Your OS: Ubuntu 18.04
- Your Ruby version: sensu-ruby-runtime 
- Your version of Puppet: 6.9 Open source
- Your version of Sensu: 5.13.2-6204
- Your version of this module:  3.8

**Comment by maxadamo** [comment]
```puppet
# == Class: myorganization_sensu::server::checks
#
#
class myorganization_sensu::server::checks (
  Stdlib::Absolutepath $gem_path   = $myorganization_sensu::params::gem_path,
  Stdlib::Absolutepath $check_path = $myorganization_sensu::params::check_path,
  Array $haproxy_backends          = $myorganization_sensu::params::haproxy_backends,
  Array $ssl_url_check             = $myorganization_sensu::params::ssl_url_check,
  Array $snmp_agent_check          = $myorganization_sensu::params::snmp_agent_check,
  Hash $ping_target_check          = $myorganization_sensu::params::ping_target_check,
  Array $domain_expiration_check   = $myorganization_sensu::params::domain_expiration_check,
  String $puppet_env               = $myorganization_sensu::params::puppet_env
) inherits myorganization_sensu::params {

  resources { 'sensu_check': purge => true; }

  sensu_check {
    default:
      ensure        => present,
      interval      => 60,
      subscriptions => ['linux'],
      handlers      => ['devops_email', 'slack'],
      require       => Class['sensu::backend'];
    # Linux subscription
    'check-authentication':
      command  => "${check_path}/check-pbis.rb {{.labels.domain_password}}",
      interval => 240;
    'check-chrony':
      command => "${check_path}/check-chrony.rb";
    'check-cpu':
      command             => "${gem_path}/sensu-plugins-cpu-checks-4.0.0/bin/check-cpu.sh -w 85 -c 95",
      low_flap_threshold  => '15',
      high_flap_threshold => '40',
      handlers            => ['devops_email_3rd_time', 'slack_3rd_time'];
    'check-disk':
      command => "sudo ${check_path}/check-disk-usage.rb -x iso9660 -w 85 -c 90";
    'check-free-ram':
      command             => "${check_path}/check-ram.rb -w 15 -c 8",
      low_flap_threshold  => '15',
      high_flap_threshold => '40',
      handlers            => ['devops_email_3rd_time', 'slack_3rd_time'];
    'check-fstab-mounts':
      command => "${check_path}/check-fstab-mounts.rb";
    'check-memory-percent':
      ensure  => absent,
      command => "${check_path}/check-memory-percent.rb -w 80 -c 90";
    'check-puppet-disabled':
      command => "${check_path}/check-puppet-disabled.rb";
    'check-puppet-last-run':
      command => "sudo ${check_path}/check-puppet-last-run.rb";
    'check-sshd':
      command => "${check_path}/check-process.rb --pattern sshd --critical-under 1";
    'metrics-cpu':
      command              => "${check_path}/metrics-cpu-pcnt-usage.rb -s {{.labels.host_name}}",
      handlers             => 'influxdb',
      output_metric_format => 'graphite_plaintext';
    'metrics-disk':
      command              => "${check_path}/metrics-disk-capacity.rb -s {{.labels.host_name}}",
      interval             => 120,
      handlers             => 'influxdb',
      output_metric_format => 'graphite_plaintext';
    'metrics-memory':
      command              => "${check_path}/metrics-memory-percent.rb -s {{.labels.host_name}}",
      handlers             => 'influxdb',
      output_metric_format => 'graphite_plaintext';
    # virtual and physical subscription
    'check-temperature':
      subscriptions => ['physical'],
      command       => "${check_path}/check_temp.pl -w 70 -c 80 -s \"Core \\d|temp\\d\"";
    # HA-Proxy subscription
    'check-haproxy':
      command       => "${check_path}/check-haproxy.rb -S /var/run/haproxy.sock -A",
      subscriptions => ['haproxy'];
    # Galera subscription
    'check-galera':
      command       => "${check_path}/check_galera_cluster -u monitor -p {{.labels.monitor_password | default \"monitor\"}}",
      subscriptions => ['galera'];
    # Consul subscription
    'check-consul-leader':
      command       => "${check_path}/check-consul-leader.rb --server localhost --port 8500",
      subscriptions => ['consul'];
    'check-consul-all':
      command       => "${check_path}/check-service-consul.rb --consul http://localhost:8500 --all",
      subscriptions => ['consul'];
    # Postgres subscription
    'check-postgres-connection':
      command       => "${check_path}/check_postgres.pl -H localhost -u pgres_check --dbpass={{.labels.monitor_password | default \"password\"}} -db pgres_check --action=connection",
      subscriptions => ['postgres'];
    # Redis subscription
    'check-redis-info':
      command       => "${check_path}/check-redis-info.rb -P {{.labels.redis_token | default \"password\"}}",
      subscriptions => ['redis'];
    # K8s subscription
    'check-k8s-services':
      command       => "${check_path}/check-kube-service-available.rb -s https://127.0.0.1:6443 -l {{.labels.k8s_service | default \"kubernetes\"}}",
      handlers      => ['devops_email', 'nmaas_email'],
      subscriptions => ['k8s-master'];
    'check-k8s-pods-running':
      command       => "${check_path}/check-kube-pods-running.rb -s https://127.0.0.1:6443",
      handlers      => ['devops_email', 'nmaas_email'],
      subscriptions => ['k8s-master'];
    'check-k8s-nodes-ready':
      command       => "${check_path}/check-kube-nodes-ready.rb -s https://127.0.0.1:6443",
      handlers      => ['devops_email', 'nmaas_email'],
      subscriptions => ['k8s-master'];
    'metrics-pods':
      command              => "${check_path}/metrics-pods.sh {{.labels.k8s_entity | default \"default\"}} ${check_path}",
      handlers             => 'influxdb',
      output_metric_format => 'influxdb_line',
      subscriptions        => ['k8s-master'];
  }

  # == Metrics HA-Proxy: this metrics goes to uat and prod only
  #
  $haproxy_backends.each | $haproxy_backend | {
    sensu_check { "metrics-haproxy-${haproxy_backend}":
      ensure               => present,
      interval             => 60,
      require              => Class['sensu::backend'],
      command              => "${check_path}/metrics-haproxy.sh ${haproxy_backend} ${puppet_env}-haproxy ${check_path}",
      handlers             => 'influxdb',
      output_metric_format => 'graphite_plaintext',
      subscriptions        => ['haproxy'];
    }
  }

  # == Certificates expiration check
  #
  $ssl_url_check.each | $check_url | {
    sensu_check {
      default:
        ensure        => present,
        handlers      => ['devops_email'],
        round_robin   => true,
        require       => Class['sensu::backend'],
        subscriptions => ['sensu_server'];
      "${check_url}_check":
        interval          => 3600,
        proxy_entity_name => "ssl_check_${check_url}",
        command           => "${check_path}/check-https-cert.rb -k -c 2 -w 5 -u https://${check_url}";
      "${check_url}_ssllabs_check":
        interval          => 7200,
        proxy_entity_name => "ssl_check_ssllabs_${check_url}",
        command           => "${check_path}/check_ssl_cert -H ${check_url} --check-ssl-labs 'B' --ignore-exp";
    }
  }

  # == Ping test
  #
  keys($ping_target_check).each | $ping_host | {
    sensu_check { "${ping_host}_check":
      ensure            => present,
      proxy_entity_name => "ping_${ping_host}",
      interval          => 60,
      round_robin       => true,
      handlers          => ['devops_email'],
      require           => Class['sensu::backend'],
      command           => "${check_path}/check-ping.rb -r -h ${ping_target_check[$ping_host]}",
      subscriptions     => ['sensu_server'];
    }
  }

  # == SNMP disk checks
  #
  $snmp_agent_check.each | $agent | {
    sensu_check { "check_disk_${agent}":
      ensure        => present,
      interval      => 120,
      handlers      => ['devops_email'],
      require       => Class['sensu::backend'],
      command       => "${check_path}/check-snmp-disk.rb -h ${agent} -w 85 -c 90 -C public",
      subscriptions => ['sensu_server'];
    }
  }

  # == Domain Expiration Check: every 4 hours (otherwise we exceed the "rate limit")
  #
  $domain_expiration_check.each | $domain_item | {
    sensu_check { "check_domain_expiration_${domain_item}":
      ensure            => present,
      proxy_entity_name => "domain_expiration_${domain_item}",
      interval          => 7200,
      round_robin       => true,
      handlers          => ['devops_email'],
      require           => Class['sensu::backend'],
      command           => "${check_path}/check_domain.sh -d ${domain_item} -z",
      subscriptions     => ['sensu_server'];
    }
  }

}
```

**Comment by maxadamo** [comment]
this is the terminal output:
```
root@uat-sensu01:~# puppet agent -t
Info: Using configured environment 'uat'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for uat-sensu01.myorganization.org
Info: Applying configuration version '1569404499'
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-authentication in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-chrony in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-consul-all in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-consul-leader in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-cpu in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-disk in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-free-ram in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-fstab-mounts in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-galera in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-haproxy in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-nodes-ready in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-pods-running in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-services in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-postgres-connection in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-puppet-disabled in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-puppet-last-run in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-redis-info in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-sshd in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-temperature in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-cpu in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-disk in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-haproxy-idp_ssl in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-haproxy-wiki_ssl in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-memory in default]/ensure: removed
Notice: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-pods in default]/ensure: removed
Error: sensuctl create check-k8s-services in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-k8sh3h' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-services): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-services]/ensure: change from 'absent' to 'present' failed: sensuctl create check-k8s-services in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-k8sh3h' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-services): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-k8s-pods-running in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-9524j2' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-pods-running): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-pods-running]/ensure: change from 'absent' to 'present' failed: sensuctl create check-k8s-pods-running in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-9524j2' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-pods-running): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-k8s-nodes-ready in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-6f0hvc' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-nodes-ready): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-k8s-nodes-ready]/ensure: change from 'absent' to 'present' failed: sensuctl create check-k8s-nodes-ready in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-6f0hvc' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-k8s-nodes-ready): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-cpu in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-q57iyk' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-cpu): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-cpu]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-cpu in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-q57iyk' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-cpu): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-disk in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-wu1o8r' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-disk): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-disk]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-disk in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-wu1o8r' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-disk): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-memory in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1bxquol' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-memory): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-memory]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-memory in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1bxquol' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-memory): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-pods in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1uqn91b' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-pods): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-pods]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-pods in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1uqn91b' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-pods): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-haproxy-wiki_ssl in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-ywgdle' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-haproxy-wiki_ssl): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-haproxy-wiki_ssl]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-haproxy-wiki_ssl in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-ywgdle' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-haproxy-wiki_ssl): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create metrics-haproxy-idp_ssl in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-qf5eur' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-haproxy-idp_ssl): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[metrics-haproxy-idp_ssl]/ensure: change from 'absent' to 'present' failed: sensuctl create metrics-haproxy-idp_ssl in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-qf5eur' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/metrics-haproxy-idp_ssl): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-authentication in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1bvrgr3' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-authentication): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-authentication]/ensure: change from 'absent' to 'present' failed: sensuctl create check-authentication in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1bvrgr3' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-authentication): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-chrony in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-p7kd7w' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-chrony): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-chrony]/ensure: change from 'absent' to 'present' failed: sensuctl create check-chrony in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-p7kd7w' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-chrony): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-disk in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1vfl4zs' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-disk): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-disk]/ensure: change from 'absent' to 'present' failed: sensuctl create check-disk in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1vfl4zs' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-disk): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-fstab-mounts in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-16v6mxc' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-fstab-mounts): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-fstab-mounts]/ensure: change from 'absent' to 'present' failed: sensuctl create check-fstab-mounts in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-16v6mxc' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-fstab-mounts): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-puppet-disabled in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1m0ybyi' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-puppet-disabled): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-puppet-disabled]/ensure: change from 'absent' to 'present' failed: sensuctl create check-puppet-disabled in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1m0ybyi' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-puppet-disabled): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-puppet-last-run in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-qfc43h' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-puppet-last-run): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-puppet-last-run]/ensure: change from 'absent' to 'present' failed: sensuctl create check-puppet-last-run in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-qfc43h' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-puppet-last-run): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-sshd in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-rgnxfw' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-sshd): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-sshd]/ensure: change from 'absent' to 'present' failed: sensuctl create check-sshd in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-rgnxfw' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-sshd): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-temperature in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-mgr17t' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-temperature): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-temperature]/ensure: change from 'absent' to 'present' failed: sensuctl create check-temperature in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-mgr17t' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-temperature): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-haproxy in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1hilwc0' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-haproxy): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-haproxy]/ensure: change from 'absent' to 'present' failed: sensuctl create check-haproxy in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1hilwc0' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-haproxy): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-galera in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1boalqn' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-galera): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-galera]/ensure: change from 'absent' to 'present' failed: sensuctl create check-galera in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1boalqn' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-galera): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-consul-leader in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-179c1pg' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-consul-leader): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-consul-leader]/ensure: change from 'absent' to 'present' failed: sensuctl create check-consul-leader in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-179c1pg' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-consul-leader): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-consul-all in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-18wukw7' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-consul-all): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-consul-all]/ensure: change from 'absent' to 'present' failed: sensuctl create check-consul-all in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-18wukw7' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-consul-all): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-postgres-connection in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1gmla6c' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-postgres-connection): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-postgres-connection]/ensure: change from 'absent' to 'present' failed: sensuctl create check-postgres-connection in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-1gmla6c' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-postgres-connection): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-redis-info in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-7kfq61' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-redis-info): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-redis-info]/ensure: change from 'absent' to 'present' failed: sensuctl create check-redis-info in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-7kfq61' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-redis-info): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-cpu in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-15j0vln' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-cpu): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-cpu]/ensure: change from 'absent' to 'present' failed: sensuctl create check-cpu in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-15j0vln' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-cpu): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Error: sensuctl create check-free-ram in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-yqv7k4' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-free-ram): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed
Error: /Stage[main]/myorganization_sensu::Server::Checks/Sensu_check[check-free-ram]/ensure: change from 'absent' to 'present' failed: sensuctl create check-free-ram in default failed
Error message: Execution of '/usr/bin/sensuctl create --file /tmp/sensuctl20190925-16984-yqv7k4' returned 1: error validating resource 0 (/api/core/v2/namespaces/default/checks/check-free-ram): check interval must be greater than 0 or a valid cron schedule must be provided
Error: resource validation failed (corrective)
Info: Class[myorganization_sensu::Server::Checks]: Unscheduling all events on Class[myorganization_sensu::Server::Checks]
Notice: Applied catalog in 30.36 seconds
root@uat-sensu01:~# 
```

**Comment by treydock** [comment]
The issue has to do with the fact when we build up the resources in Puppet we do `"#{name} in #{namespace}"`. For resource matching we match on actual name + namespace but it seems the `resources` type isn't using the prefetch matching, just matching on literal resource name.  I am trying to see if there's a way to solve this issue but for now one work around would be to append ` in default` to your `sensu_check` resources so `check-authentication` would become `check-authentication in default`. This will fix the issue your facing and allow you to still enable purging.

The complicated name handling is necessary to support things like a check with same name in different namespaces.

**Comment by maxadamo** [comment]
ah! If that's the way it's supposed to work, I am fine with it. I tested it and it's working. 
I simply didn't know, that the namespace had to be put in the resource title.... and if it's not present it defaults to default.

**Comment by treydock** [comment]
The way you had done it before was how I intended , requiring the composite name of `#{name} in #{namespace}` was not something I intended. I am working on a new type called `sensu_resources` that will be used to replace `resources` so you'd do `sensu_resources { 'sensu_check': purge => true }` and could leave namespace off the checks like you had originally done. The Puppet provided `resources` type isn't doing what we need for name matching.
